src/iuse.cpp

```diff
@@ -3809,6 +3809,58 @@ int iuse::mp3_on(player *p, item *it, bool t, const tripoint &pos)
     return it->type->charges_to_use();
 }
 
+int iuse::gasmask( player *p, item *it, bool t, const tripoint &pos )
+{
+    if( t ) { // Normal use
+        if( p->is_worn( *it ) ) {
+            // calculate amount of absorbed gas per filter charge
+            const field &gasfield = g->m.field_at( pos );
+            for( auto &dfield : gasfield ) {
+                const field_entry &entry = dfield.second;
+                const field_id fid = entry.getFieldType();
+                switch( fid ) {
+                    case fd_smoke:
+                        it->set_var( "gas_absorbed", it->get_var( "gas_absorbed", 0 ) + 12 );
```

Comment:
Should move to field definitions json later, so we get rid of hardcode.

Reply:
Good idea, this can be used for custom fields.